internal/iam/command_ipgroup_update.go

```diff
@@ -0,0 +1,134 @@
+package iam
+
+import (
+	iamv2 "github.com/confluentinc/ccloud-sdk-go-v2/iam/v2"
+	"github.com/spf13/cobra"
+	"slices"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+)
+
+func (c *ipGroupCommand) newUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update <id>",
+		Short: "Update an IP Group.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.update,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Update the Group Name of IP Group "ipg-12345"":`,
+				Code: `confluent iam ip-group update ipg-12345 --group_name "New Group Name"`,
+			},
+		),
+	}
+
+	pcmd.AddProviderFlag(cmd, c.AuthenticatedCLICommand)
+	cmd.Flags().String("group_name", "", "Name of the IP Group.")
+	cmd.Flags().StringSlice("cidr_blocks", []string{}, "List of CIDR blocks to replace existing CIDR blocks on the IP Group.")
+	cmd.Flags().StringSlice("add_cidr_blocks", []string{}, "List of CIDR blocks to add to existing CIDR blocks on the IP Group.")
+	cmd.Flags().StringSlice("remove_cidr_blocks", []string{}, "List of CIDR blocks to remove from existing CIDR blocks on the IP Group.")
+
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddFilterFlag(cmd)
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *ipGroupCommand) update(cmd *cobra.Command, args []string) error {
+	flags := []string{
+		"group_name",
+		"cidr_blocks",
+		"add_cidr_blocks",
+		"remove_cidr_blocks",
+	}
+	if err := errors.CheckNoUpdate(cmd.Flags(), flags...); err != nil {
+		return err
+	}
+
+	groupName, err := cmd.Flags().GetString("group_name")
+	if err != nil {
+		return err
+	}
+
+	cidrBlocks, err := cmd.Flags().GetStringSlice("cidr_blocks")
+	if err != nil {
+		return err
+	}
+
+	addCidrBlocks, err := cmd.Flags().GetStringSlice("add_cidr_blocks")
+	if err != nil {
+		return err
+	}
+
+	removeCidrBlocks, err := cmd.Flags().GetStringSlice("remove_cidr_blocks")
+	if err != nil {
+		return err
+	}
+
+	currentIpGroupId := args[0]
+
+	// get the current IP group we are going to be updating
+	currentIpGroup, err := c.V2Client.GetIamIpGroup(currentIpGroupId)
+	// initialize our new cidr blocks with the existing values
+	newCidrBlocks := currentIpGroup.GetCidrBlocks()
+
+	if err != nil {
+		return err
+	}
```

Comment:
Does this block only apply in the case where we are add/remove (not overwrite?)

Reply:
In the situation where we overwrite, `newCidrBlocks` gets it's value re-set to `cidrBlocks`.

In the situation where we add/remove, we modify `newCidrBlocks` with the specified elements.

There's a way to do it where we separate it for the two options, but this way felt more visually clear to me in the code.